From: Antoine Musso <hashar@free.fr>
Date: Fri, 7 Sep 2018 23:50:16 +0200
Subject: wmf: ignore submit error on merged change

Sometime changes get manually merged. Attempting to submit it would
cause Gerrit to error out with: error: fatal: change is merged

Ignore Gerrit review exception when using --submit and the errors has
'change is merged'.

Bug: T203846
Change-Id: I775fd1d632fa647b2b7b20c69057e5d7949a8c35
---
 zuul/connection/gerrit.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/zuul/connection/gerrit.py b/zuul/connection/gerrit.py
index 95bf35c..b3a30e0 100644
--- a/zuul/connection/gerrit.py
+++ b/zuul/connection/gerrit.py
@@ -279,7 +279,14 @@ class GerritConnection(BaseConnection):
             else:
                 cmd += ' --%s %s' % (key, val)
         cmd += ' %s' % change
-        out, err = self._ssh(cmd)
+
+        # wmf: ignore submit error on merged change T203846
+        try:
+            out, err = self._ssh(cmd)
+        except Exception:
+            if 'submit' in action and err and 'change is merged' in err:
+                return
+            raise
         return err
 
     def query(self, query):
